affinity/cpp-23/d1436r3.md

```diff
@@ -298,18 +302,20 @@ The `bulk_execution_affinity_t` property provides nested property types and obje
 |----------------------|----------------------|--------------|
-| `bulk_execution_affinity_t::none_t` | `bulk_execution_affinity_t::none` | A call to `e.bulk_execute(f, s, sf)` has no requirements on the binding of *execution agents* to the underlying *execution resources*. |
-| `bulk_execution_affinity_t::scatter_t` | `bulk_execution_scatter_t::scatter` | A call to `e.bulk_execute(f, s, sf)` should aim to bind the created *execution agents* to the underlying *execution resources* (ordered by physical closeness) such that they are distributed equally across the *execution resources* in a round-robin fashion. <br><br> If the execution context associated with `e` is not able to bind the *execution agents* to the underlying *execution resources* as requested it should fall back to `bulk_execution_affinity_t::none_t`. |
-| `bulk_execution_affinity_t::compact_t` | `bulk_execution_compact_t::compact` | A call to `e.bulk_execute(f, s, sf)` should aim to bind the created *execution agents* to the underlying *execution resources* such that they are distributed as close as possible to the *execution resource* of the *thread of execution* which created them. <br><br> If the execution context associated with `e` is not able to bind the *execution agents* to the underlying *execution resources* as requested it should fall back to `bulk_execution_affinity_t::none_t`. |
-| bulk_execution_affinity_t::balanced_t | bulk_execution_balanced_t::balanced | A call to `e.bulk_execute(f, s, sf)` should aim to bind the created *execution agents* to the underlying *execution resources* (ordered by physical closeness) such that they are distributed equally across the *execution resources* in a bin packing fashion. <br><br> If the execution context associated with `e` is not able to bind the *execution agents* to the underlying *execution resources* as requested it should fall back to `bulk_execution_affinity_t::none_t`. |
+| `bulk_execution_affinity_t::none_t` | `bulk_execution_affinity_t::none` | A call to `execution::bulk_execute(e, f, s)` is not required to bind the created *execution agents* for the work-items of the iteration space specified by `s` to *execution resources*. |
+| `bulk_execution_affinity_t::spread_t` | `bulk_execution_spread_t::spread` | A call to `e.bulk_execute(f, s, sf)` should aim to bind the created *execution agents* for the work-items of the iteration space specified by `s` to *execution resources* such that the average locality distance of adjacent work-items in the same subdivision of the available concurrency is maximized and the average locality distance of adjacent work-items in different subdivisions of the available concurrency is maximized. Only binding subsequent *execution agents* to an *execution resource* if no other *execution resources* would otherwise result in fewer *execution agents* being bound to it. <br><br> If `e` is not able to fulfil this aim the it should fall back to `bulk_execution_affinity_t::none_t`. |
```

Comment:
This has now been updated to the following wording:

> The binding of all execution agents to all execution resources must not result in the difference between the number of execution agents assigned to any execution resources being greater than 1.

I have also added some preamble to introduce the terms of art "locality distance" and "available concurrency":

> The `bulk_execution_affinity_t` nested properties are defined using the following terms of art:
> * *Available concurrency*; which refers to the number of *execution resources* available to an *executor* which can be bound to *execution agents* concurrently, assuming no contention.
> * *Locality distance*; which refers to an implementation-defined metric for measuring the relative affinity between *execution resources* whereby *execution resources* with a lower *locality distance* are likely to have similar latency in memory access operations,  for a given memory location.

> The `bulk_execution_affinity_t` nested properties also refer to the subdivision of *execution resources*, which is an implementation-defined method of subdividing the *available concurrency*, generally based on groupings of *execution resources* with the lowest *locality distance* to each other.

> [*Note:* An alternative term of art for *locality distance* could be *locality interference*. *--end note*]

Reply:
1. Instead of "which refers to," you could just say "is" -- it's a definition.
2. I like the "available concurrency" definition. :-)
3. "Locality distance" makes sense but lower latency might not necessarily always be the goal.  For example, some kinds of memory might have higher latency for individual accesses, but higher bandwidth for many accesses with a particular access pattern.  The latter makes it the "preferred memory" for some kinds of processors.  Maybe we could fudge this by saying "performance"?

Reply:
Thanks for the feedback, I've changed the statements as you suggested and I've updated the definition for "locality distance" to also incorporate bandwidth.